scripts/tests/chiptest/test_definition.py

```diff
@@ -159,10 +159,20 @@ def __terminateProcess(self):
         if self.process:
             self.process.terminate()  # sends SIGTERM
             try:
-                self.process.wait(10)
+                exit_code = self.process.wait(10)
+                # Ideally, we should raise an exception for all non-zero exit codes, but some
+                # applications (notably the tv-* apps) are not built with proper SIGTERM
```

Comment:
can we fix these apps instead?

Reply:
@andy31415 We can try, but it's a pretty large amount of work best done by the people who understand those apps.  I don't think we should block this PR on it.

Reply:
I would also prefer if these apps were fixed, especially since they have code paths that likely aren’t exercised by other apps, making them ideal for leak checks. However, as Boris pointed out, fixing them would require significant effort and is better addressed by the developers familiar with this particular use case. This PR, on the other hand, aims to stop leaks in all the other, more commonly used, cases.

Reply:
Spent a little bit more time. https://github.com/project-chip/connectedhomeip/pull/35843 is just moving one line of code around and fix the crashes...